tvx/extract_msg.go

```diff
@@ -161,6 +184,16 @@ func runExtractMsg(c *cli.Context) error {
 		return err
 	}
 
+	// don't fetch additional content that wasn't accessed yet during car spidering / generation.
+	type onlineblockstore interface {
+		SetOnline(bool)
+	}
+	if ob, ok := pst.Blockstore.(onlineblockstore); ok {
+		ob.SetOnline(false)
+	}
```

Comment:
This is cool. Can we add a CLI flag to control this behaviour so we can generate both coarse-grained and fine-grained vectors?

```sh
$ tvx extract-msg --state-retain=[accessed-actors/accessed-cids]
```

Reply:
Actually, I think this is not good enough for the fine-grained monitoring. Instead, we need to record which CIDs are effectively accessed (blockstore.Get()) during the execution of a transaction, and only write those out to the CAR.

Reply:
what do you think is getting included that shouldn't be?

Reply:
I think this will include all CIDs that both happened to be linked AND happen to be in our local store by this point (e.g. as a result of executing precursors, or whatever), even if they are not accessed by _this_ message. This is also a bit non-deterministic I suspect.

The goal of fine-grained vectors is to trim state strictly to only include the CIDs that are effectively accessed during the processing of a message, keeping all other leaves/nodes as broken links.

Reply:
So basically what we want to do is add `StartTracing()` and `FinishTracing() []cid.Cid` methods to our proxying blockstore to record and obtain which CIDs are read.